### PR TITLE
Add recipe for flycheck-swift3

### PR DESCRIPTION
**Pull request by gooichi**
### Brief summary of what the package does

Apple swift-mode support for Flycheck.

### Direct link to the package repository

https://github.com/GyazSquare/flycheck-swift3

### Your association with the package

I am the maintainer.

### Relevant communications with the upstream package maintainer

None needed

### Checklist

- [x] I've read [CONTRIBUTING.md](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.md)
- [x] I've used [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] I've built and installed the package using the instructions in the [README](https://github.com/melpa/melpa/blob/master/README.md)

### CODE REVIEW

File: recipes/flycheck-swift3 [hunk 1]
```diff
@@ -0,0 +1,3 @@
+(flycheck-swift3 :fetcher github
+                 :repo "GyazSquare/flycheck-swift3"
+                 :branch "master")
```
Comment (syohex) [inline recipes/flycheck-swift3]: You need not to specify `:branch` attribute if it is default branch.

**Comment by syohex** [comment]
Its package description is `Flycheck: Swift support`. This is similar to [flycheck-swift](https://github.com/swift-emacs/flycheck-swift). I suppose it should make the differences clear.

File: recipes/flycheck-swift3 (hunk 1 above)
Comment (gooichi) [inline recipes/flycheck-swift3]: I removed `:branch` attribute.

**Comment by gooichi** [comment]
Thank you for your comment!

I'll improve the package description ASAP.

**Comment by gooichi** [comment]
I improved the description (a little).

**Comment by syohex** [comment]
Sorry my explanation about package description was unclear. If https://github.com/swift-emacs/flycheck-swift cannot work with Swift 3 or higher versions, package description of this package should contain `Swift 3` for users.

**Comment by gooichi** [comment]
Flycheck-swift3 is designed to work with Apple's official [swift-mode.el](https://github.com/apple/swift/blob/master/utils/swift-mode.el). The name `flycheck-swift3`comes from requiring Apple Swift compiler 3.0 or later. If you set `flycheck-swift3` to use the tools for 2.x (the toolchain option), you can use swift 2.x.

I don't know [flycheck-swift](https://github.com/swift-emacs/flycheck-swift) well, sorry... :-(

**Comment by syohex** [comment]
I see. Could you write it into Commentary section and README.md ? I suppose users cannot understand such details from current documentation.

**Comment by gooichi** [comment]
Thank you for your reply.

OK, I'll improve README.md. Then I contact you again on this request.

**Comment by syohex** [comment]
Thanks, I'm waiting for you response.

**Comment by gooichi** [comment]
I have updated the Commentary section and README.md.
Could you confirm them?

**Comment by syohex** [comment]
Merged

**Comment by gooichi** [comment]
Thank you very much! :-)

**Comment by purcell** [comment]
Thanks @gooichi & @syohex !

**Pull request merged**